sdk/servicebus/service-bus/samples/typescript/src/advanced/sessionManagerSample.ts

```diff
@@ -0,0 +1,155 @@
+    // actually be obtained. This goes away when we have `await createSessionReceiver()`.
+    await sessionReceiver.getState();
+  } catch (err) {
+    // TODO: when this happens do I still need to close the session receiver? (again, temporary concern)
+    if (_noSessionsAvailable(err)) {
+      console.log(`INFO: no available sessions, sleeping for ${delayWhenNoSessionsAvailableMs}`);
+      await delay(delayWhenNoSessionsAvailableMs);
+      return;
+    }
+
+    throw err;
+  }
+
+  await handlers.processInitialize(sessionReceiver.sessionId);
+
+  const idleTimer = _createIdleTimer(sessionIdleTimeoutMs);
+  let threwError = false;
+
+  sessionReceiver.subscribe(
+    {
+      async processMessage(msg) {
+        idleTimer.refresh();
+        await handlers.processMessage(msg);
+      },
+      async processError(err) {
+        threwError = true;
+        await handlers.processError(err, sessionReceiver.sessionId);
+      }
+    },
+    {
+      abortSignal
+    }
+  );
+
+  await idleTimer.expirationPromise;
+
+  try {
+    await sessionReceiver.close();
+  } catch (err) {
+    await handlers.processError(err, sessionReceiver.sessionId);
+  }
+
+  await handlers.processClose(threwError ? "error" : "idle_timeout", sessionReceiver.sessionId);
+}
+
+async function _roundRobinThroughAvailableSessions(
+  handlers: SessionMessageHandlers,
+  abortSignal: AbortSignalLike
+): Promise<void> {
+  const serviceBusClient = new ServiceBusClient(serviceBusConnectionString);
+  const allPromises = [];
+
+  for (let i = 0; i < maxSessionsToProcessSimultaneously; ++i) {
+    allPromises.push(
+      new Promise(async (_res, rej) => {
+        try {
+          while (!abortSignal.aborted) {
+            await _processNextSession(serviceBusClient, handlers, abortSignal);
+          }
+        } catch (err) {
+          rej(err);
+        }
+      })
+    );
+  }
+
+  // if we get a fatal error in any of the promises we can just shut it all down
+  console.log(`All session promises have been started`);
+  await Promise.race(allPromises);
+  console.log(`Exiting...`);
+}
+
+async function main() {
+  const abortController = new AbortController();
+
+  const handlers: SessionMessageHandlers = {
+    async processInitialize(sessionId) {
+      console.log(`[${sessionId}] will start processing...`);
+    },
+    async processMessage(msg) {
+      console.log(`[${msg.sessionId}] received message with body ${msg.body}`);
+    },
+    async processError(err, sessionId) {
+      console.log(`[${sessionId}] had error ${err.message}`);
+    },
+    async processClose(reason, sessionId) {
+      if (reason === "error") {
+        console.log(`[${sessionId}] was closed because of error`);
+      } else if (reason === "idle_timeout") {
+        console.log(`[${sessionId}] was closed - no more messages within idle timeout`);
+      }
+    }
+  };
+
+  await _roundRobinThroughAvailableSessions(handlers, abortController.signal);
```

Comment:
What is the reason to wait for the promise returned by `_roundRobinThroughAvailableSessions` to resolve?
Looks like if there was no fatal error from individual sessions, then this promise would never resolve.

Reply:
The .push() and all of that is gone. We just let the promises float free.